dot/network/status.go

```diff
@@ -36,22 +37,27 @@ const ExpireStatusInterval = SendStatusInterval + time.Minute
 type status struct {
 	host          *host
 	hostMessage   *StatusMessage
-	peerConfirmed map[peer.ID]time.Time
-	peerMessage   map[peer.ID]*StatusMessage
+	peerConfirmed *sync.Map //map of peer.ID to time.Time
+	peerMessage   *sync.Map //map of peer.ID to *StatusMessage
 }
 
 // newStatus creates a new status instance from host
 func newStatus(host *host) *status {
 	return &status{
 		host:          host,
-		peerConfirmed: make(map[peer.ID]time.Time),
-		peerMessage:   make(map[peer.ID]*StatusMessage),
+		peerConfirmed: &sync.Map{},
+		peerMessage:   &sync.Map{},
 	}
 }
 
 // confirmed returns true if peer is confirmed
 func (status *status) confirmed(peer peer.ID) bool {
-	return !status.peerConfirmed[peer].IsZero()
+	t, ok := status.peerConfirmed.Load(peer)
+	if !ok {
```

Comment:
this could be simplified in a oneliner.

Reply:
not really, since t is used again at line 60, it needs to be defined outside of the if statement